components/camel-bindy/src/main/java/org/apache/camel/dataformat/bindy/csv/BindyCsvDataFormat.java

```diff
@@ -138,58 +144,74 @@ public Object unmarshal(Exchange exchange, InputStream inputStream) throws Excep
         // List of Pojos
         List<Map<String, Object>> models = new ArrayList<>();
 
-        // Pojos of the model
-        Map<String, Object> model;
         InputStreamReader in = null;
-        Scanner scanner = null;
         try {
             if (checkEmptyStream(factory, inputStream)) {
                 return models;
             }
     
             in = new InputStreamReader(inputStream, IOHelper.getCharsetName(exchange));
-    
-            // Scanner is used to read big file
-            scanner = new Scanner(in);
-    
+
             // Retrieve the separator defined to split the record
             String separator = factory.getSeparator();
             String quote = factory.getQuote();
             ObjectHelper.notNull(separator, "The separator has not been defined in the annotation @CsvRecord or not instantiated during initModel.");
-    
-            int count = 0;
-            
-            // If the first line of the CSV file contains columns name, then we
-            // skip this line
-            if (factory.getSkipFirstLine()) {
-                // Check if scanner is empty
-                if (scanner.hasNextLine()) {
-                    scanner.nextLine();
+            AtomicInteger count = new AtomicInteger(0);
+
+            // Use a Stream to stream a file across.
+            try (Stream<String> lines = new BufferedReader(in).lines()) {
```

Comment:
Could this be tied up to an option on data format config? (Considering existing users)

Reply:
I'm not entirely clear what you mean by this; could you elaborate?

Reply:
you can define a boolean flag in org.apache.camel.dataformat.bindy.BindyAbstractDataFormat and implement this new feature if it's set to true / leave it as-is if no option provided / false.

other data format types like BindyFixedLengthDataFormat and BindyKeyValuePairDataFormat may require similar implementation. Could you also have a look at them?

Reply:
Good call on the other classes; I'll take a look at those and follow up with another commit to address those deficiencies.  A quick skim suggests that they too would suffer from the same bug.

As for adding a boolean option to address this - I'm not entirely convinced of the value of it.  The assumption the code is making is that an entire line from the file is being read, but it leaves the definition of a line to the implementation of `Scanner`, when it really should be the convention we all have come to expect and demand of lines from a file (carriage return, line feed, or both).  By and large, `Scanner` does catch almost every use case there is when it comes to lines, with the exception of this very peculiar instance.

Reply:
yes i agree, but this is for backward compatibility. some parts may be dropped or cleaned up in camel 3 or maybe your change may be only introduced in 2.23 unless earlier versions would require the same thing. @oscerd wdyt?

Reply:
I'm okay with that approach.

I'm still working through this - I believe the fixed format is giving me some fits right now - but I do hope to have this wrapped up soon.

Reply:
@MakotoTheKnight any update on your work?

Reply:
@davsclaus I did reach a stumbling block with BindyFixedLengthDataFormat - it seems to read in files in a completely different fashion.  I plan to shore up  BindyKeyValuePairDataFormat very soon - hopefully within the next couple of days.